app/modules/annotations/models.py

```diff
@@ -126,6 +135,124 @@ def get_job_debug(self, verbose):
                 log, f'Annotation {self.guid} not connected to an encounter'
             )
 
+    @classmethod
+    def get_sage_sync_tags(cls):
+        return 'annotation', 'annot'
+
+    def sync_with_sage(
+        self, ensure=False, force=False, bulk_sage_uuids=None, skip_asset=False, **kwargs
+    ):
+        from app.extensions.sage import (
+            to_sage_uuid,
+            from_sage_uuid,
+            SAGE_UNKNOWN_NAME,
+        )
+
+        # First, ensure that the annotation's asset has been synced with Sage
+        if not skip_asset:
+            self.asset.sync_with_sage(
+                ensure=ensure, force=force, bulk_sage_uuids=bulk_sage_uuids, **kwargs
+            )
+
+            if self.asset.content_guid is None:
+                log.error(
+                    'Asset for Annotation %r failed to send, cannot send annotation to Sage'
+                    % (self,)
+                )
+                # We tried to sync the asset's content GUID, but that failed... it is likely that the asset's file is missing
+                return
+
+        if force:
+            with db.session.begin(subtransactions=True):
+                self.content_guid = None
+                db.session.merge(self)
+            db.session.refresh(self)
+
+        if ensure:
+            if self.content_guid is not None:
+                if bulk_sage_uuids is not None:
+                    # Existence checks can be slow one-by-one, bulk checks are better
+                    if self.content_guid in bulk_sage_uuids.get('annotation', {}):
+                        # We have found this Annotation on Sage, simply return
+                        return
+                else:
+                    content_guid_str = str(self.content_guid)
+                    sage_rowids = current_app.sage.request_passthrough_result(
+                        'annotation.exists', 'get', args=content_guid_str, target='sync'
+                    )
+                    if len(sage_rowids) == 1:
```

Comment:
If we're ever doing a reformatting PR later, these two ifs could be a single if ___ and ___. Pretty minor code style thing so not blocking this PR.

Reply:
Consolidated